app/uk/gov/hmrc/gform/controllers/AuthenticatedRequestActions.scala

```diff
@@ -121,10 +121,16 @@ class AuthenticatedRequestActions(
     f: Request[AnyContent] => LangADT => AuthCacheWithoutForm => Future[Result]): Action[AnyContent] = Action.async {
     implicit request =>
       implicit val l: LangADT = getCurrentLanguage(request)
+
       for {
         formTemplate <- gformConnector.getFormTemplate(formTemplateId)
         authResult <- authService
-                       .authenticateAndAuthorise(formTemplate, request.uri, getAffinityGroup, ggAuthorised(request))
+                       .authenticateAndAuthorise(
+                         formTemplate,
+                         request.uri,
+                         getAffinityGroup,
+                         ggAuthorised(request),
+                         request.cookies.get(appConfig.`case-worker-assumed-identity-cookie`))
```

Comment:
Sorry for being pain in the ass, but can we have a function which given a `request` will return `assumedIdentity`, so we can reuse it at both places this line is currently duplicated.

Reply:
OK

Reply:
I've factored this out now into a function